test/metabase/test/data/interface.clj

```diff
@@ -213,21 +213,33 @@
   Currently, this only affects `db-qualified-table-name`."
   nil)
 
+(defn- normalize-qualified-name [n]
+  (-> n u/lower-case-en (str/replace #"-" "_")))
+
+(defn- db-qualified-table-name-prefix [db-name]
+  (str (normalize-qualified-name (or *database-name-override* db-name))
+       \_))
+
+(defn qualified-by-db-name?
+  "Is `table-name` qualified by the name of its database? See [[db-qualified-table-name]] for more details."
+  [db-name table-name]
+  (str/starts-with? table-name (db-qualified-table-name-prefix db-name)))
+
 (defn db-qualified-table-name
   "Return a combined table name qualified with the name of its database, suitable for use as an identifier.
   Provided for drivers where testing wackiness makes it hard to actually create separate Databases, such as Oracle,
   where this is disallowed on RDS. (Since Oracle can't create seperate DBs, we just create various tables in the same
-  DB; thus their names must be qualified to differentiate them effectively.)"
+  DB; thus their names must be qualified to differentiate them effectively.)
+
+  Asserts that the resulting name has fewer than 30 characters, because databases like Oracle have limits on the
+  lengths of identifiers."
   ^String [^String database-name, ^String table-name]
-  {:pre [(string? database-name) (string? table-name)]}
-  ;; take up to last 30 characters because databases like Oracle have limits on the lengths of identifiers
-  (-> (or *database-name-override* database-name)
-      (str \_ table-name)
-      u/lower-case-en
-      (str/replace #"-" "_")
-      (->>
-        (take-last 30)
-        (apply str))))
+  {:pre [(string? database-name)
+         (string? table-name)]
+   :post [(qualified-by-db-name? database-name %)
+          (<= (count %) 30)]}
```

Comment:
This appears to have broken the Presto tests:

```
 ERROR in metabase.query-processor-test.timezones-test/native-sql-params-filter-test (interface.clj:228)
Uncaught exception, not in assertion.

java.lang.AssertionError: Assert failed: (<= (count %) 30)
                 metabase.test.data.interface/db-qualified-table-name       interface.clj:  228
```

Reply:
Ah see you figured this out already 😄

Reply:
Yeah, got a bit confident with asking for a review before all tests passed